kvbc/include/kvbc_adapter/state_snapshot_adapter.hpp

```diff
@@ -0,0 +1,73 @@
+// Concord
+//
+// Copyright (c) 2022-2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <string>
+#include <type_traits>
+
+#include "state_snapshot_interface.hpp"
+#include "categorization/kv_blockchain.h"
+#include "rocksdb/native_client.h"
+#include "blockchain_type.hpp"
+#include "ReplicaConfig.hpp"
+
+using concord::storage::rocksdb::NativeClient;
+using concord::kvbc::IKVBCStateSnapshot;
+using concord::kvbc::KVBlockChain;
+using concord::kvbc::categorization::Converter;
+
+namespace concord::kvbc::statesnapshot {
+class KVBCStateSnapshot : public IKVBCStateSnapshot {
+ public:
+  template <typename OPTIONS,
```

Comment:
Why do we need an extra class for the state snapshot? can't it be part of KeyValueBlockchain ?